src/transformers/models/mvp/modeling_mvp.py

```diff
@@ -0,0 +1,2049 @@
+# coding=utf-8
+# Copyright 2022 The Fairseq Authors and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" PyTorch MVP model."""
+import copy
+import math
+import random
+from typing import List, Optional, Tuple, Union
+
+import torch
+import torch.utils.checkpoint
+from torch import nn
+from torch.nn import BCEWithLogitsLoss, CrossEntropyLoss, MSELoss
+
+from ...activations import ACT2FN
+from ...modeling_outputs import (
+    BaseModelOutput,
+    BaseModelOutputWithPastAndCrossAttentions,
+    CausalLMOutputWithCrossAttentions,
+    Seq2SeqLMOutput,
+    Seq2SeqModelOutput,
+    Seq2SeqQuestionAnsweringModelOutput,
+    Seq2SeqSequenceClassifierOutput,
+)
+from ...modeling_utils import PreTrainedModel
+from ...utils import (
+    add_code_sample_docstrings,
+    add_end_docstrings,
+    add_start_docstrings,
+    add_start_docstrings_to_model_forward,
+    logging,
+    replace_return_docstrings,
+)
+from .configuration_mvp import MvpConfig
+
+
+logger = logging.get_logger(__name__)
+
+_CHECKPOINT_FOR_DOC = "RUCAIBox/mvp"
+_CONFIG_FOR_DOC = "MvpConfig"
+_TOKENIZER_FOR_DOC = "MvpTokenizer"
+
+# Base model docstring
+_EXPECTED_OUTPUT_SHAPE = [1, 8, 1024]
+
+MVP_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "RUCAIBox/mvp",
+    "RUCAIBox/mvp-data-to-text",
+    "RUCAIBox/mvp-open-dialog",
+    "RUCAIBox/mvp-question-answering",
+    "RUCAIBox/mvp-question-generation",
+    "RUCAIBox/mvp-story",
+    "RUCAIBox/mvp-summarization",
+    "RUCAIBox/mvp-task-dialog",
+    "RUCAIBox/mtl-data-to-text",
+    "RUCAIBox/mtl-multi-task",
+    "RUCAIBox/mtl-open-dialog",
+    "RUCAIBox/mtl-question-answering",
+    "RUCAIBox/mtl-question-generation",
+    "RUCAIBox/mtl-story",
+    "RUCAIBox/mtl-summarization",
+    # See all MVP models at https://huggingface.co/models?filter=mvp
+]
+
+
+def shift_tokens_right(input_ids: torch.Tensor, pad_token_id: int, decoder_start_token_id: int):
```

Comment:
Can we add `# Copied from statements ...` here? The code is more or less copied from Bart no? 
So maybe we can add statements like this: https://github.com/huggingface/transformers/blob/09178705101b9803e7b9ea7f79a46b4c242dd4bf/src/transformers/models/mbart/modeling_mbart.py#L94

Reply:
Fixed.